iOS/DuckDuckGo/Subscription/ViewModel/SubscriptionFlowViewModel.swift

```diff
@@ -75,11 +72,34 @@ final class SubscriptionFlowViewModel: ObservableObject {
         self.userScript = userScript
         self.subFeature = subFeature
         self.subscriptionManager = subscriptionManager
+
+        self.webViewSettings = AsyncHeadlessWebViewSettings(bounces: false,
+                                                            allowedDomains: Self.allowedDomains(baseURL: subscriptionManager.url(for: .baseURL),
+                                                                                                isInternalUser: isInternalUser),
+                                                            contentBlocking: false)
+
+
         self.webViewModel = AsyncHeadlessWebViewViewModel(userScript: userScript,
                                                           subFeature: subFeature,
                                                           settings: webViewSettings)
     }
 
+    // Allowed domains
+    private static func allowedDomains(baseURL: URL, isInternalUser: Bool) -> [String] {
+        var allowedDomains = Set<String>()
+
+        // Allow navigation to baseURLs domain
+        allowedDomains.insert(baseURL.host ?? "duckduckgo.com")
+
+        // For internal user allow these domains as required for DUO based authentication flow
+        if isInternalUser {
+            allowedDomains.formUnion(["use-login.duckduckgo.com", "duosecurity.com", "login.microsoftonline.com"])
+        }
+
+        assert(!allowedDomains.isEmpty, "Allowed domains should not be empty.")
+        return Array(allowedDomains)
+    }
+
```

Comment:
One thought here is whether we should make test this logic, whether via directly by making the function ‘internal’ or indirectly via calling another method on the view model.

Reply:
Good point, I have updated the PR.